app/src/main/kotlin/compiler/lowlevel/linearization/IFTPattern.kt

```diff
@@ -51,8 +51,8 @@ sealed class IFTPattern {
 
     // Matches every intermediate form tree node
     // Doesn't consume the nodes, returns it as a subtree
-    class AnyNode : IFTPattern() {
-        override fun match(node: IFTNode): MatchResult? {
+    data class AnyNode(val dummy: Unit = Unit) : IFTPattern() {
```

Comment:
Why the `dummy` arg?

Reply:
Because we do not want bare `class`es when other cases in a `sealed class` are `data class`es, and a `data class` needs at least one field. This trick satisfies this requirement.

Reply:
Also, replacing such cases with `object`s is generally not the best idea, because sometimes we want to distinguish by identity. In case of `IFTPattern` this shouldn't be a problem (I only changed this for consistency), because we do not use it with `Ref`, but for example `IFTNode.Dummy` definitely should not be an `object`.